File: src/main/kotlin/com/huytran/rerm/controller/UserController.java
//package com.huytran.rerm.controller;
//
//import com.huytran.rerm.bean.core.BeanResult;
//import com.huytran.rerm.controller.core.CoreController;
//import com.huytran.rerm.service.UserService;
//import org.springframework.web.bind.annotation.RestController;
//
//import javax.servlet.http.HttpSession;
//import javax.validation.Valid;
//
//@RestController("/user")
//public class UserController extends CoreController<UserService, UserService.Params> {
//
//    private UserService userService;
//
//    public UserController(
//            UserService userService
//    ) {
//        super(userService);
//        this.userService = userService;
//    }
//
//    public BeanResult login(
//            HttpSession httpSession,
//            @Valid UserService.Params params) {
//        return userService.login(httpSession, params);
//    }
//
//    public BeanResult signup(
//            HttpSession httpSession,
//            @Valid UserService.Params params) {
//        return userService.signup(httpSession, params);
//    }
//
//}

File: src/main/kotlin/com/huytran/rerm/controller/UserController.kt
package com.huytran.rerm.controller

import com.huytran.rerm.bean.core.BeanBasic
import com.huytran.rerm.service.UserService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/user")
class UserController(private val userService: UserService) {

    @PostMapping("/create")
    fun create(
            @RequestParam(name = "username") username: String,
            @RequestParam(name = "password") password: String,
            @RequestParam(name = "name") name: String,
            @RequestParam(name = "address") address: String,
            @RequestParam(name = "phoneNumber") phoneNumber: String): BeanBasic {
        return userService.create(
                username,
                password,
                name,
                address,
                phoneNumber
        )
    }

    @PostMapping("/delete")
    fun delete(
            @RequestParam(name = "id") id: Long) {
        userService.delete(id)
    }

    @PostMapping("/getNativeDetail")
    fun getNativeDetail(): List<Any> {
        return userService.getNativeDetail()
    }

}